sdks/python/apache_beam/runners/interactive/display/pcoll_visualization.py

```diff
@@ -0,0 +1,269 @@
+
+from apache_beam import pvalue
+from apache_beam.runners.interactive import interactive_environment as ie
+from apache_beam.runners.interactive import pipeline_instrument as instr
+
+try:
+  import jsons  # pylint: disable=import-error
+  from IPython import get_ipython  # pylint: disable=import-error
+  from IPython.core.display import HTML  # pylint: disable=import-error
+  from IPython.core.display import Javascript  # pylint: disable=import-error
+  from IPython.core.display import display  # pylint: disable=import-error
+  from IPython.core.display import display_javascript  # pylint: disable=import-error
+  from IPython.core.display import update_display  # pylint: disable=import-error
+  from facets_overview.generic_feature_statistics_generator import GenericFeatureStatisticsGenerator  # pylint: disable=import-error
+  from timeloop import Timeloop  # pylint: disable=import-error
+
+  if get_ipython():
+    _pcoll_visualization_ready = True
+  else:
+    _pcoll_visualization_ready = False
+except ImportError:
+  _pcoll_visualization_ready = False
+
+# 1-d types that need additional normalization to be compatible with DataFrame.
+_one_dimension_types = (int, float, str, bool, list, tuple)
+
+_DIVE_SCRIPT_TEMPLATE = """
+            document.querySelector("#{display_id}").data = {jsonstr};"""
+_DIVE_HTML_TEMPLATE = """
+            <script src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/1.3.3/webcomponents-lite.js"></script>
+            <link rel="import" href="https://raw.githubusercontent.com/PAIR-code/facets/1.0.0/facets-dist/facets-jupyter.html">
+            <facets-dive sprite-image-width="{sprite_size}" sprite-image-height="{sprite_size}" id="{display_id}" height="600"></facets-dive>
+            <script>
+              document.querySelector("#{display_id}").data = {jsonstr};
+            </script>"""
+_OVERVIEW_SCRIPT_TEMPLATE = """
+              document.querySelector("#{display_id}").protoInput = "{protostr}";
+              """
+_OVERVIEW_HTML_TEMPLATE = """
+            <script src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/1.3.3/webcomponents-lite.js"></script>
+            <link rel="import" href="https://raw.githubusercontent.com/PAIR-code/facets/1.0.0/facets-dist/facets-jupyter.html">
+            <facets-overview id="{display_id}"></facets-overview>
+            <script>
+              document.querySelector("#{display_id}").protoInput = "{protostr}";
+            </script>"""
+_DATAFRAME_PAGINATION_TEMPLATE = """
+            <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script> 
+            <script src="https://cdn.datatables.net/1.10.16/js/jquery.dataTables.js"></script> 
+            <link rel="stylesheet" href="https://cdn.datatables.net/1.10.16/css/jquery.dataTables.css">
+            {dataframe_html}
+            <script>
+              $("#{table_id}").DataTable();
+            </script>"""
+
+
+def visualize(pcoll, dynamic_plotting_interval=None):
```

Comment:
Yes, I'll mark it as experimental! Actually the whole module is marked as `For internal use only; no backwards-compatibility guarantees.`

For renaming, we only need to touch the `visualize()` function that has not been implemented in module `interactive_beam`. Since a notebook user should only use APIs provided in that module.

Reply:
SGTM.